frontend/lib/search/results_loader.dart

```diff
@@ -102,7 +102,7 @@ class ResultsLoaderState extends State<ResultsLoader> {
         // Params are outdated.
         // If we're still at the first key, we must manually retrigger fetching.
         if (pageKey == _pagingController.firstPageKey) {
-          return _fetchPage(pageKey);
+          return await _fetchPage(pageKey);
         }
       }
       _pagingController.error = error;
```

Comment:
To fix the mentioned issue, we can probably do an early return when `!this.mounted` is true after every `await`. (But should probably be done in a separate PR :D )

Reply:
Yea, I saw this approach but haven't tested it yet